src/differentials/abstract_differential.jl

```diff
@@ -11,18 +11,23 @@ All subtypes of `AbstractDifferential` implement the following operations:
 
 `+(a, b)`: linearly combine differential `a` and differential `b`
 
-`*(a, b)`: multiply the differential `a` by the differential `b`
+`*(a, b)`: multiply the differential `b` by the scalaring factor `b`
 
 `Base.conj(x)`: complex conjugate of the differential `x`
 
-`extern(x)`: convert `x` into an appropriate non-`AbstractDifferential` type for
-use outside of `ChainContext`.
+`Base.zero(x) = Zero()`: a zero.
 
-Valid arguments to these operations are `T` where `T<:AbstractDifferential`, or
-where `T` has proper `+` and `*` implementations.
+In general a differential is the type of a derivative of a value.
+The type of the value is for contrast called the primal type.
+Differntial types correspond to primal types, though the relation is not one-to-one.
+Not only `AbstractDifferential` are valid differentials.
```

Comment:
I find this way round easier to read (at first i thought the other one was sayin "not all `AbsDifferentials` are valid differentials" and was much confused.)

```suggestion
Some valid differentials are not subtypes of `AbstractDifferential`.
```

Reply:
How about:
```suggestion
Subtypes of  `AbstractDifferential` are not the only differential types.
```

Reply:
like it